node/core/approval-voting/src/lib.rs

```diff
@@ -633,23 +675,21 @@ async fn run<C>(
 ) -> SubsystemResult<()>
 	where C: SubsystemContext<Message = ApprovalVotingMessage>
 {
-	let (background_tx, background_rx) = mpsc::channel::<BackgroundRequest>(64);
 	let mut state = State {
 		session_window: RollingSessionWindow::new(APPROVAL_SESSIONS),
 		keystore: subsystem.keystore,
 		slot_duration_millis: subsystem.slot_duration_millis,
 		db: ApprovalDBV1Reader::new(subsystem.db.clone(), subsystem.db_config.clone()),
 		clock,
 		assignment_criteria,
+		approvals_cache: lru::LruCache::new(128usize),
+		candidate_hash_map: HashMap::new(),
+		currently_checking: BTreeMap::new(),
 	};
 
 	let mut wakeups = Wakeups::default();
 
-	// map block numbers to background work.
-	let mut background_tasks = BTreeMap::new();
```

Comment:
If you refactor the cache & currently-checking-set out of the `State` struct, they can be local variables just like this was and `wakeups` is - cleaner.

Reply:
Just saw this. Adding a commit to address.